tiledb/cf/creator.py

```diff
@@ -61,11 +61,36 @@ class DataspaceCreator:
 
     def __init__(self):
         self._dims: MutableMapping[str, SharedDim] = {}
-        self._array_creators: Dict[str, ArrayCreator] = {}
+        self._array_creators: List[Optional[ArrayCreator]] = []
+        self._array_name_to_id: Dict[str, int] = {}
         self._dim_to_arrays: Dict[str, List[str]] = defaultdict(list)
         self._attr_to_array: Dict[str, str] = {}
         self._attr_dataspace_names: Dict[str, str] = {}
 
+    def __iter__(self):
+        """Iterators over all array creators."""
+        for array_creator in self._array_creators:
+            if array_creator is not None:
+                yield array_creator
+
+    def __getitem__(self, array_name: str) -> ArrayCreator:
```

Comment:
`DataspaceCreator` contains several pieces of information apart from array creators. Is it "obvious" to a user of this class that `list(dataspace_creator)` or `dataspace_creator["foo"]` return array creators (and not, say, attr creators)?